frontend/src/app/category-view/category-view.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { Category, CategoryBackendService } from "../services/category-backend.service";
+import { KeycloakService } from "keycloak-angular";
+import { ActivatedRoute } from "@angular/router";
+import { Item, ItemBackendService } from "../services/item-backend.service";
+import { MatTable } from "@angular/material/table";
+
+@Component({
+  selector: 'app-category-view',
+  templateUrl: './category-view.component.html',
+  styleUrls: ['./category-view.component.css']
+})
+export class CategoryViewComponent implements OnInit {
+
+  id: number = this.route.snapshot.params['id'];
```

Comment:
`this.route.snapshot.params` is bad approach, because it requires to reload page.

Better to use `this.route.paramMap` where `route: ActivatedRoute`
```
constructor(
    private route: ActivatedRoute,
) {
}

data: Category;
loading: boolean; // this variable is used on html to show that data is loading - you can show spinner when loading is true, then show data when loading is finished

ngOnInit() {
  this.loading = true;
  this.route.paramMap
      .pipe(
          takeUntil(this.unsubscribe),  // unsubscribe every subscribe() function, read more on internet. bigger example in other comment
          switchMap((params: ParamMap) => {
                  const id = params.get('id');
                  return this.categoryService.getCategoryById(Number(id));
              }
          )
      )
      .subscribe((data) => {
          this.data = data;
      })
      .add(() => {
          this.loading = false;
      });
}
```